GW2EIUpdater/Updater.cs

```diff
@@ -0,0 +1,177 @@
+    public const string EI_DownloadName = "GW2EI.zip";
+    public const string EICLI_DownloadName = "GW2EICLI.zip";
+    public const string EI_TempFolder = "GW2EIUpdateTemp";
+    public const string EIUpdater_Executable = "GuildWars2EliteInsightsUpdater.exe";
+    private readonly string Executable = Process.GetCurrentProcess().ProcessName;
+    private const string LatestReleaseURL = "https://api.github.com/repos/baaron4/GW2-Elite-Insights-Parser/releases/latest";
+    private readonly HttpClient httpClient = new();
+    private readonly bool DownloadCLI = false;
+    private GitHubRelease LatestRelease;
+
+    public Updater()
+    {
+    }
+
+    public Updater(bool downloadCLI)
+    {
+        DownloadCLI = downloadCLI;
+    }
+
+    /// <summary>
+    /// Checks the version of the latest GitHub release.
+    /// </summary>
+    /// <returns>Returns true if the latest version has a higher number than current.</returns>
+    public async Task<bool> NewReleaseCheckerAsync()
+    {
+        var assembly = Assembly.GetEntryAssembly();
+
+        if (assembly != null)
+        {
+            Version currentVersion = assembly.GetName().Version!;
+
+            // GitHub API Call & JSON Object creation
+            try
+            {
+                var uri = new Uri(LatestReleaseURL);
+                httpClient.DefaultRequestHeaders.UserAgent.ParseAdd("GW2EI-Updater/1.0");
+                httpClient.DefaultRequestHeaders.Accept.ParseAdd("application/vnd.github+json");
+                using var responseMessage = await httpClient.GetAsync(uri);
+                responseMessage.EnsureSuccessStatusCode();
+                var jsonResponse = await responseMessage.Content.ReadAsStringAsync();
+                LatestRelease = JsonConvert.DeserializeObject<GitHubRelease>(jsonResponse);
+            }
+            catch
+            {
+                LatestRelease = null;
+            }
+
+            if (LatestRelease != null && currentVersion != null)
+            {
+                // Release format is "v1.0.0.0"
+                string version = LatestRelease.Name.Substring(1); // Remove "v"
+                var latestVersion = Version.Parse(version);
+
+                // Store release information
+                CurrentVersion = currentVersion.ToString();
+                LatestVersion = latestVersion.ToString();
+                ReleasePageURL = LatestRelease.Html_Url;
+                UpdateFound = latestVersion > currentVersion;
+            }
+            return UpdateFound;
+        }
+        
+        return UpdateFound;
+    }
+
+    public async Task DownloadFileAsync()
+    {
+        string downloadUrl = string.Empty;
+        
+        // Windows: C:\Users\User\AppData\Local\Temp\
+        // Linux: /tmp/
+        string tempPath = Path.GetTempPath();
+
+        // Windows: C:\Users\User\AppData\Local\Temp\GW2EIUpdateTemp\
+        // Linux: /tmp/GW2EIUpdateTemp/
+        string folderPath = Path.Combine(tempPath, EI_TempFolder);
```

Comment:
This should not be required. The temp path is by definition already a temporary folder for you to throw files in.
If you want to avoid collisions then there is [GetTempFileName](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=net-9.0)